components/automate-ui/src/app/entities/layout/layout.facade.ts

```diff
@@ -53,15 +57,24 @@ export class LayoutFacadeService {
   public contentHeight = `calc(100% - ${Height.Navigation}px)`;
   public sidebar$: Observable<MenuItemGroup[]>;
   public showPageLoading$: Observable<boolean>;
+  private authorizedChecker: AuthorizedChecker;
 
   constructor(
+    fullStore: Store<NgrxStateAtom>,
     private store: Store<fromLayout.LayoutEntityState>,
     private layoutSidebarService: LayoutSidebarService
   ) {
     this.sidebar$ = store.select(sidebar);
     this.showPageLoading$ = store.select(showPageLoading);
     this.updateDisplay();
-    this.store.dispatch(new GetProjects());
+
+    this.authorizedChecker = new AuthorizedChecker(fullStore);
+    this.authorizedChecker.setPermissions([{ endpoint: '/iam/v2/projects', verb: 'get' }], []);
+    this.authorizedChecker.isAuthorized$
+      .pipe(filter(identity), first())
+      .subscribe(
+        // wait for permission and, if it comes at all, fire exactly once
+        () => this.store.dispatch(new GetProjects()));
```

Comment:
This GetProjects call occurs during page construction on application startup/page refresh.
Note that the `pipe` operators are crucial here.
First, `filter(identity)` translates to "wait until endpoint `/iam/v2/projects` has method `get` set to `true`. That, will never happen for a user with no permissions. And that is OK. But for a user with permission, it will happen in fairly short order. Once it does, invoke `GetProjects` and then _guarantee we never do it again_! That is handled by the `first()` call   which completes the observable, thereby effectively subscribing, grabbing a single value, and unsubscribing automatically. (Without the `first()`, if you happen to be on the ingest rules tab of a project page, it will flicker out-of-control.)

Reply:
I've forgotten why this GetProjects is fired here in the first place. To populate the project filter?